### PR TITLE
Flip the default go image from alpine/musl to wolfi/glibc!

### PR DESCRIPTION
**Pull request by dlorenc**
The old one is still available but at cgr.dev/chainguard/go:musl-latest.

This was generated by copying the old latest-glibc.apko.yaml -> latest.apko.yaml and moving the old latest.apko.yaml -> latest-musl.apko.yaml (the diff looks a little weird).

Signed-off-by: Dan Lorenc <dlorenc@chainguard.dev>

### CODE REVIEW

**Review by luhring — APPROVED** [review]
This is a breaking change, but it also seems inevitable

File: images/go/configs/latest-musl.apko.yaml [hunk 1]
```diff
@@ -19,13 +20,8 @@ accounts:
...
       gid: 65532
+  # TODO: this should run as 65532
   run-as: 0
-
 entrypoint:
   command: /usr/bin/go
-
 cmd: help
-
-archs:
```
Comment (imjasonh) [inline images/go/configs/latest-musl.apko.yaml:29]: We're going to lose a lot of archs if we do this. Not sure if folks are using them though, without some more querying.

Comment (dlorenc) [reply images/go/configs/latest-musl.apko.yaml:29]: Ack. I'm personally fine with it for these images. Happy to hold if we want to wait longer though.

**Comment by kimsterv** [comment]
> This is a breaking change, but it also seems inevitable

Who are we breaking and how should we be communicating these changes? cc @amouat

**Comment by amouat** [comment]
> > This is a breaking change, but it also seems inevitable
> 
> Who are we breaking and how should we be communicating these changes? cc @amouat

I'm not sure. I think it's pretty limited in breaking people. I'm assuming it compiles against the system libs and therefore you get a glibc binary instead of the previous musl one, which may not work in certain environments.

It will break people using archs other than arm64 and amd64, but I suspect there's so few of those we can ignore it.

I would make the change but add something to the page explaining the change and how to get the musl image.

**Comment by dlorenc** [comment]
> > > This is a breaking change, but it also seems inevitable
> > 
> > 
> > Who are we breaking and how should we be communicating these changes? cc @amouat
> 
> I'm not sure. I think it's pretty limited in breaking people. I'm assuming it compiles against the system libs and therefore you get a glibc binary instead of the previous musl one, which may not work in certain environments.
> 
> It will break people using archs other than arm64 and amd64, but I suspect there's so few of those we can ignore it.
> 
> I would make the change but add something to the page explaining what happened and how to get the musl image.

Added a sentence to the page.

**Pull request merged**